domain/secretbackend/state/state.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	coredatabase "github.com/juju/juju/core/database"
+	coremodel "github.com/juju/juju/core/model"
+	coresecrets "github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/domain"
+	modelerrors "github.com/juju/juju/domain/model/errors"
+	"github.com/juju/juju/domain/secretbackend"
+	backenderrors "github.com/juju/juju/domain/secretbackend/errors"
+)
+
+// Logger is the interface used by the state to log messages.
+type Logger interface {
+	Debugf(string, ...interface{})
+	Warningf(string, ...interface{})
+}
+
+// State represents database interactions dealing with secret backends.
+type State struct {
+	*domain.StateBase
+	logger Logger
+}
+
+// NewState returns a new secret backend state based on the input database factory method.
+func NewState(factory coredatabase.TxnRunnerFactory, logger Logger) *State {
+	return &State{
+		StateBase: domain.NewStateBase(factory),
+		logger:    logger,
+	}
+}
+
+// GetModel is responsible for returning the model with the provided uuid.
+func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (*coremodel.Model, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	stmt, err := sqlair.Prepare(`
+SELECT 
+    model_uuid AS &Model.uuid,
+    name       AS &Model.name,
+    type       AS &Model.type
+FROM v_model_metadata
+WHERE model_uuid = $M.uuid`, sqlair.M{}, Model{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var m Model
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, sqlair.M{"uuid": uuid}).Get(&m)
+		if errors.Is(err, sql.ErrNoRows) {
+			return fmt.Errorf("model %q:%w", uuid, modelerrors.NotFound)
```

Comment:
So this will read "model xxxx: model not found" which is a bit meh.
We really want `fmt.Errorf("model %q not found%w", uuid, errors.Hide(modelerrors.NotFound))`